io-engine/src/grpc/v1/replica.rs

```diff
@@ -492,4 +588,103 @@ impl ReplicaRpc for ReplicaService {
         )
         .await
     }
+    #[named]
+    async fn list_replica_snapshot(
+        &self,
+        request: Request<ListReplicaSnapshotsRequest>,
+    ) -> GrpcResult<ListReplicaSnapshotsResponse> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                info!("{:?}", args);
+                let rx = rpc_submit(async move {
+                    // if replica_uuid is valid, filter snapshot based on
+                    // replica_uuid
+                    if !args.replica_uuid.is_empty() {
```

Comment:
hmm instead we should make replica_uuid optional to make it clear that not providing the uuid is a way to get all snapshots.